src/main/java/com/theelixrlabs/healthcare/service/PatchDoctorService.java

```diff
@@ -33,14 +32,15 @@ public PatchDoctorService(DoctorRepository doctorRepository, Validator validator
     /**
      * Replace existing doctor details based on request.
      *
-     * @param doctorId  Doctor unique UUID
-     * @param doctorDto Data transfer object containing doctor information.
+     * @param doctorId    Doctor unique UUID as a String
+     * @param doctorDto    Data transfer object containing doctor information.
      * @return The replaced doctor details.
      * @throws CustomException If validation fails or doctor already exists.
      */
-    public DoctorDto patchDoctorById(UUID doctorId, DoctorDto doctorDto) throws CustomException {
+    public DoctorDto patchDoctorById(String doctorId, DoctorDto doctorDto) throws CustomException {
+        UUID validDoctorId = validator.validateAndConvertToUUID(doctorId, MessageConstants.INVALID_UUID);
         validator.validateDoctor(doctorDto);
-        Optional<DoctorModel> optionalDoctor = doctorRepository.findById(doctorId);
+        Optional<DoctorModel> optionalDoctor = doctorRepository.findById(validDoctorId);
         if (optionalDoctor.isPresent()) {
             DoctorModel existingDoctor = optionalDoctor.get();
             if (doctorDto.getFirstName() != null && !doctorDto.getFirstName().equals(existingDoctor.getFirstName())) {
```

Comment:
Here also you can call the hasValueForUpdate method. If the method is called across multiple classes, you can move that private function to a common util class.

Reply:
added to utility class